packages/webapp/src/containers/CropVarieties/useCropVarietyCatalogue.js

```diff
@@ -0,0 +1,262 @@
+import {
+  getCurrentManagementPlans,
+  getExpiredManagementPlans,
+  getPlannedManagementPlans,
+} from '../managementPlanSlice';
+import { useSelector } from 'react-redux';
+import { cropCatalogueFilterDateSelector, cropVarietyFilterSelector } from '../filterSlice';
+import { useMemo } from 'react';
+import useStringFilteredCrops from '../CropCatalogue/useStringFilteredCrops';
+import {
+  ACTIVE,
+  COMPLETE,
+  LOCATION,
+  PLANNED,
+  STATUS,
+  SUPPLIERS,
+  NEEDS_PLAN,
+} from '../Filter/constants';
+import { useTranslation } from 'react-i18next';
+import useFilterNoPlan from './useFilterNoPlan';
+import useSortByCropTranslation from '../CropCatalogue/useSortByCropTranslation';
+import { managementPlansWithCurrentLocationSelector } from '../Task/TaskCrops/managementPlansWithLocationSelector';
+import { cropVarietiesSelector } from '../../containers/cropVarietySlice';
+
+export default function useCropVarietyCatalogue(filterString, crop_id) {
```

Comment:
Is there any way to reduce the number of loops and array methods you use in this function? By my count there are at least 22 loops over various arrays of data using various forms of `Array.map()`, `Array.filter()`, and explicit for loops (this is excluding calls to various functions like `useFilterNoPlan` which I am relatively sure also loop over the data. This makes it really confusing to follow exactly what is happening with the various sets of data and how they are being manipulated to their final form. Either adding some comments explaining the loops and/or reducing the number of loops would be very helpful.

Reply:
Added comments on the top of each function/ implementation